### PR TITLE
PD-6750 Public cookbook for warden

### PR DESCRIPTION
**Pull request by rvanburen-r7**
adds a public cookbook to warden in the cookbook directory

### CODE REVIEW

File: cookbook/libraries/helpers.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+end
+
+##
+# Version and Download URI helpers
+##
+module Tokend
+  module Helpers
+    class << self
+      def github_download(owner, repo, version)
+        "https://github.com/#{owner}/#{repo}/releases/download/v#{version}/tokend_#{version}_amd64.deb"
```
Comment (dgreene-r7) [inline cookbook/libraries/helpers.rb]: `warden_#{version}_amd64.deb`?

File: cookbook/metadata.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
+## Use package.json as the source of truth
+project_path = ::File.expand_path('../../', __FILE__)
+package_dot_json = JSON.parse(IO.read(::File.join(project_path, 'package.json')))
+
+name package_dot_json['name']
+description 'Install and configure https://github.com/rapid7/tokend'
```
Comment (dgreene-r7) [inline cookbook/metadata.rb]: Warden?

File: cookbook/libraries/helpers.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  ##
+  def cookbook_version
+    run_context.cookbook_collection[cookbook_name].version
+  end
+end
+
+##
+# Version and Download URI helpers
+##
+module Tokend
```
Comment (dgreene-r7) [inline cookbook/libraries/helpers.rb]: Warden

File: cookbook/recipes/default.rb [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  owner node['warden']['user']
+  group node['warden']['group']
+  mode '0755'
+
+  recursive true
+end
+
+## Fetch and install warden
+remote_file 'warden' do
+  source warden::Helpers.github_download('rapid7', 'warden', node['warden']['version'])
```
Comment (dgreene-r7) [inline cookbook/recipes/default.rb]: Should be `Warden::Helpers` once you change the module name in `cookbook/libraries/helpers.rb`.

**Comment by dgreene-r7** [comment]
This cookbook is missing a way to actually put the Warden configuration on disk. See https://github.com/rapid7/propsd/blob/master/cookbook/recipes/default.rb#L90-L98 for an example.

**Comment by jmanero-r7** [comment]
👍

**Pull request merged**